cert-provider/cert-provider.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/sh
+
+ACME_BIN="$(realpath ~/.acme.sh/acme.sh)"
+
+# colour output helpers...
+reset=$(tput -T xterm sgr0)
+red=$(tput -T xterm setaf 1)
+green=$(tput -T xterm setaf 2)
+yellow=$(tput -T xterm setaf 3)
+blue=$(tput -T xterm setaf 4)
+
+function logError() {
+    echo "${red}[Error]${reset} $1"
+}
+
+function logWarn() {
+    echo "${yellow}[Warn]${reset} $1"
+}
+
+function logInfo() {
+    echo "${blue}[Info]${reset} $1"
+}
+
+function logSuccess() {
+    echo "${green}[Success]${reset} $1"
+}
+
+# join domains with -d separator
+join_domains()  { echo "$1" | sed -e "s/,/ -d /g" -e "s/^/-d /g"; }
+
+DO_CERTS="true"
+if [ -z "$ACTIVE" ]; then
+    logError "ACTIVE must be set. Value can be anything."
+    unset DO_CERTS
+fi
+
+if [ -z "$DOMAINS" ]; then
+    logError "DOMAINS must be set. Value should be a comma-delimited string of domains."
+    unset DO_CERTS
+else
+    ACME_DOMAINS=$(join_domains "$DOMAINS")
+fi
+
+if [ -z "$EMAIL" ]; then
+    logError "EMAIL must be set. Value should be a valid email address."
+    unset DO_CERTS
+fi
+
+if [ -z "$VALIDATION" ]; then
+    logInfo "VALIDATION not set. Using default: http-01"
+    VALIDATION="http-01"
+else
+    case "$VALIDATION" in
+        "http-01")
+            logInfo "Using validation method: $VALIDATION"
+            ;;
+        "dns-01")
+            logError "The DNS validation method is not supported at this time."
+            unset DO_CERTS
+            ;;
+        *)
+            logError "VALIDATION is invalid. Use a valid value: http-01"
+            unset DO_CERTS
+            ;;
+    esac
+fi
+
+if [ -z "$OUTPUT_PEM" ]; then
+    logError "OUTPUT_PEM must be set. Value should be the path to install your certificate to."
+    unset DO_CERTS
+fi
+
+ACME_OPTS=""
+
+LAST_RUN_PRODUCTION="$(cat /usr/src/app/certs/last_run_production)"
+if [ "$LAST_RUN_PRODUCTION" != "$PRODUCTION" ]; then
+    logInfo "PRODUCTION mode changed; forcing renewal..."
+    ACME_OPTS="$ACME_OPTS --force"
+fi
+echo "$PRODUCTION" > /usr/src/app/certs/last_run_production
+        
+case "$PRODUCTION" in
+    "true"|"yes")
+        logInfo "Using PRODUCTION mode"
+        ;;
+    *)
+        logInfo "Using STAGING mode"
+        ACME_OPTS="$ACME_OPTS --staging"
+        ;;
+esac
+
+case "$VALIDATION" in
+    "http-01")
+        ACME_OPTS="$ACME_OPTS --standalone"
+        ;;
+    *)
+        logError "VALIDATION is invalid. Use a valid value: http-01 or dns-01"
+        unset DO_CERTS
+        ;;
+esac
+
+if [ ! -z "$DO_CERTS" ]; then
+    FIRST_DOMAIN="${DOMAINS%%,*}"
+    logInfo "Waiting for $FIRST_DOMAIN to be available..."
+    until $(curl --output /dev/null --silent --head --fail http://$FIRST_DOMAIN); do
```

Comment:
I don't think you want the `$()` here

```suggestion
    until curl --output /dev/null --silent --head --fail "http://$FIRST_DOMAIN"; do
```